woocommerce/google-listings-and-ads
# Edit Paid Ads Campaign

### Changes proposed in this Pull Request:



- Part of issue #161 . 
- Uses Ads Campaign API from PR https://github.com/woocommerce/google-listings-and-ads/pull/141. 
- Contains changes from PR https://github.com/woocommerce/google-listings-and-ads/pull/423, please review PR https://github.com/woocommerce/google-listings-and-ads/pull/423 first before this one. 

This PR adds the ability to edit paid ads campaign. From the dashboard page, users can click on the edit button for paid ads campaign to go to the edit page. Users can only change the daily average cost amount. Upon clicking the save changes button, the button will show a spinner icon, and then redirects back to the dashboard page with updated value in the All Programs table.

### Note:

Apparently we can't disable the audience country select dropdown, it is a shortcoming of the [WooCommerce SelectControl component](https://github.com/woocommerce/woocommerce-admin/tree/main/packages/components/src/select-control). Issue https://github.com/woocommerce/woocommerce-admin/issues/6764 has been created in the WC Admin repo for this. For now, users can click on the control and interact with it, but they won't be able to change to a new value.

### Screenshots:

Demo video with my voice: https://d.pr/v/5aHfnR

Static screenshot: 

![image](https://user-images.githubusercontent.com/417342/113611722-80283b80-9681-11eb-9504-78e4ee9e2cd7.png)

### Detailed test instructions:

1. Open https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard
2. Click on the edit button for a paid ads campaign. You should see the edit campaign page with a loading spinner, and then the edit form.
3. Change the daily average cost amount. 
4. Click on the save changes button. You should see loading indicator in the button, and then redirect back to the dashboard page.
5. The amount for the paid ads campaign in the All Programs table should be the updated value.

### Changelog Note:

Add ability to edit paid ads campaign.
